### PR TITLE
Add regression tests for fruitsalad tab color fix

### LINKED ISSUE #2165: fruitsalad: first tab color not used for rest of page
From NJIT: "When clicking the first sub tab, "Learn" the body does not change colors to be purple like how clicking "Teach" turns it blue and "Volunteer" makes it green. Instead it is set to tabcolor0 and appears as if the "Learn" tab has not been clicked."

Yale's site also does this. MIT's doesn't.

### PR DESCRIPTION
**Pull request by Oval17**
## Description

When a top tab header_link and its first sub-link point to the same URL, the old logic defaulted to tabcolor0 (logo color) because the sub-link match was never strictly greater. Now defaults to tabcolor1 when the header_link is fully consumed, so the first tab gets its section color.

Adds regression tests for extract_theme covering the fix, deeper sub-tab matching, unmatched URLs, and the empty header_link edge case.

## Related Issue

Closes #2165

## Type of Change

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update
- [ ] Refactor / cleanup
- [ ] CI/CD or build change

## Testing

- [ ] Existing tests pass
- [ ] New tests added (if applicable)
- [ ] Manually verified

## Checklist

- [ ] Self-reviewed the code
- [ ] Updated documentation (if needed)
- [ ] No new warnings or errors introduced

### CODE REVIEW

**Comment by Dhirenderchoudhary** [comment]
Tests cover the main scenarios. Checks pass locally. Looks good to merge.

File: esp/esp/web/templatetags/main.py [hunk 1]
```diff
@@ -78,7 +78,10 @@ def extract_theme(url):
         num_chars_matched = count_matching_chars(url, category['header_link'])
         if num_chars_matched > max_chars_matched:
             max_chars_matched = num_chars_matched
-            tab_index = 0
+            if category['header_link'] and num_chars_matched >= len(category['header_link']) and category['links']:
```
Comment (moksha-hub) [inline esp/esp/web/templatetags/main.py]: This broadens the behavior beyond the duplicate-header-link case from #2165. As written, any URL that fully matches header_link now flips to 	abcolor1 whenever the category has at least one sub-link, even if none of those sub-links match more specifically. For example, with header_link='/learn/' and first sub-link /learn/classes/, a page like /learn/randompage now returns 	abcolor1 instead of the old 	abcolor0. If the intent is only to handle the 'header_link and first sub-link are the same URL' case, this condition needs to key off that exact overlap instead of every full header-link match.

Comment (Oval17) [reply esp/esp/web/templatetags/main.py]: I'll narrow it to:
```
if category['links'] and category['links'][0]['link'] == category['header_link']:
    tab_index = 1
```
This makes it trigger only when header_link and the first sub-link point to the same URL, which is exactly the overlap case the fix was intended for. 
Thanks for pointing out.

**Comment by Oval17** [comment]
Hi , 
The bug seems to be resolved in #5146 by tie breaking logic, 
and its already merged and working properly.
I have only kept test cases here.
If we want we can see the TCs only and close the issue by merging this PR
or if we dont want TCs right now then we can close this pr and close the issue as well
But in either case the bug is resolved and we should close the issue.
Thanks
cc : @willgearty

**Review by kkbrum — APPROVED** [review]
Thanks!
